src/image/crop.rs

```diff
@@ -374,12 +378,18 @@ impl<Cropped, Original> CropResult<Cropped, Original> {
 
     /// If the image was fully empty, crop to one single pixel of all the transparent pixels instead,
     /// leaving the layer intact while reducing memory usage.
+    ///
+    /// # Panics
+    /// Panics if the layer has zero width and height (which indicates an invalid layer state).
+    /// This should never happen with properly constructed images.
     pub fn or_crop_to_1x1_if_empty(self) -> Cropped where Original: Crop<Cropped=Cropped> + GetBounds {
         match self {
             CropResult::Cropped (cropped) => cropped,
             CropResult::Empty { original } => {
                 let bounds = original.bounds();
-                if bounds.size == Vec2(0,0) { panic!("layer has width and height of zero") }
+                if bounds.size == Vec2(0,0) {
+                    unreachable!("layer has zero width and height - this indicates an invalid layer state that should have been caught during construction")
```

Comment:
I agree that it probably won't happen - but still it should return an error if it does happen.